fhir-notification-nats/README.MD

```diff
@@ -0,0 +1,98 @@
+# fhir-notification-nats
+
+The fhir-notification-nats module is designed to publish patient records to a [NATS](http://nats.io) streaming cluster for pub/sub, so published patient records can be consumed by designated downstream applications.  
+
+## Configure IBM FHIR
+In fhir-server/liberty-config/config/default/fhir-server-config.json, find the default NATS configuration shown here:  
+
+```json
+    "nats": {
+        "enabled": false,
+        "cluster": "nats-streaming",
+        "channel": "fhirNotifications",
+        "clientId": "fhir-server",
+        "servers": "nats://nats-node1:4222,nats://nats-node2:4222,nats://nats-node3:4222",
+        "useTLS": false,
+        "truststoreLocation": "resources/security/nats.client.truststore.jks",
+        "truststorePassword": "change-password",
+        "keystoreLocation": "resources/security/nats.client.keystore.jks",
+        "keystorePassword": "change-password"
+    }
+```
+ 
+Set "enabled" to true and provide the name of your NATS cluster for the value of "cluster".  You can leave "channel" and "clientId" as defined.  Provide the URL for one or more NATS servers as the value for "servers". See the "Use TLS" section below for a discussion of the TLS settings, if required.  
+
+For use with the test configuration in the next section, set "enabled" to true and make no other changes.
+
+## Create a simple NATS cluster for testing
+
+In the directory fhir-server-test/src/test/resources/nats you will find the files to set up a basic NATS cluster and IBM FHIR Server on the same docker network.  Follow these steps to test:
+
+* Build your IBM FHIR image with the NATS configuration discussed above and publish it to Docker Hub.  Edit the fhir_server service image in fhir-server-test/src/test/resources/nats/docker-compose.yml to point to this new image.  
+
+
+```  
+fhir_server:  
+	image: <your-repo>/<your-image>  
+```  
+
+
+Note: As an alternative to building and publishing your own image, you can start an IBM FHIR Server image that contains the NATS publisher functionality, then copy the updated configuration file containing the above changes into the IBM FHIR container and restart the container.  
```

Comment:
I think we should steer users toward the "alternative" approach...no need to push to dockerhub just to test the nats stuff